### PR TITLE
Create CompatHelper.yml

### PR DESCRIPTION
**Pull request by jb-mackay**
# PULL REQUEST

## Purpose and Content
Add compat entries in Project.toml and CompatHelper workflow to maintain these.

## Benefits and Risks
Automates the maintenance of dependency compatibility.

## Linked Issues
Addresses #73 

## PR Checklist
- [x] This PR has a corresponding issue OR is linked to an SDI.
- [x] I have followed CliMA's codebase [contribution](https://clima.github.io/ClimateMachine.jl/latest/Contributing/) and [style](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) guidelines OR N/A.
- [x] I have followed CliMA's [documentation policy](https://github.com/CliMA/policies/wiki/Documentation-Policy).
- [x] I have checked all issues and PRs and I certify that this PR does not duplicate an open PR.
- [x] I linted my code on my local machine prior to submission OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code used in an integration test OR N/A.
- [x] All tests ran successfully on my local machine OR N/A.
- [x] All classes, modules, and function contain docstrings OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by jb-mackay** [comment]
Next step is to use [CompatDevTools.jl](https://github.com/charleskawczynski/CompatDevTools.jl) to generate first compat entries!

File: Project.toml [hunk 1]
```diff
@@ -7,4 +7,8 @@ version = "0.1.0"
 ClimaCore = "d414da3d-4745-48bb-8d80-42e94e092884"
 DocStringExtensions = "ffbed154-4ef7-542d-bbb7-c09d3a79fcae"
 PrettyTables = "08abe8d2-0d0c-5749-adfa-8a2ac140af0d"
-SciMLBase = "0bca4576-84f4-4d90-8ffe-ffa030f20462"
+
+[compat]
+ClimaCore = "0.10"
+DocStringExtensions = "0.9"
```
Comment (jb-mackay) [inline Project.toml:13]: v0.9 necessary to use `@template` with parametric types!

Comment (jb-mackay) [reply Project.toml:13]: This has been resolved as of Documenter v0.27.22

**Review by charleskawczynski — APPROVED** [review]
Code changes look fine to me, can you maybe just comment where the compat yaml was taken from?

**Comment by jb-mackay** [comment]
> Code changes look fine to me, can you maybe just comment where the compat yaml was taken from?

CompatHelper.yml copied from [CompatHelper.jl](https://github.com/JuliaRegistries/CompatHelper.jl/blob/master/.github/workflows/CompatHelper.yml)

**Pull request merged**